### PR TITLE
Parse query template params

### PR DESCRIPTION
**Pull request by mattwalo32**
Fixes here:
- Parse ? as a template param
- Allow ON CONFLICT clause to omit parentheses around a single target column

Only look at g4 files and base lexer.

The rest are generated.

### CODE REVIEW

File: postgresql_lexer_base.go [hunk 1]
```diff
@@ -51,6 +51,47 @@ func (receiver *PostgreSQLLexerBase) CheckLaMinus() bool {
...
+	if receiver.GetInputStream().LA(1) != '?' {
+		return false
+	}
+
+	text := receiver.GetText()
+	if len(text) == 0 {
+		return false
+	}
+
+	// My assumption is that it only makes sense to have a template param after a comparison operator
```
Comment (WorkForPizza) [inline postgresql_lexer_base.go:78]: is this assumption necessary? thinking about functions as a counterexample. `SELECT * FROM abc WHERE def = UPPER(?);`

Comment (mattwalo32) [reply postgresql_lexer_base.go:78]: Since this is inside the `Operator` rule in the lexer, the function and the comment are only in the context of matching an `Operator` token.

The lexer rule tries to group together multiple adjacent operator characters into a single operator token so that something like `>=` is a single operator, not two side-by-side operator.

So the assumption is that if a `?` is next to another operator character and that character is a comparison operator, then the `?` is a template param.

The example you've provided is a good edge case, and I've added a test for it to the bitbucket PR. It passes with the current implementation because it's a single ? alone and not next to a comparison operator.

**Review by WorkForPizza — APPROVED** [review]
